xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioDSP.h

```diff
@@ -0,0 +1,1288 @@
+    int (__cdecl* input_resample_samplerate)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE);
+
+    unsigned int (__cdecl* pre_process_needed_samplesize)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE, unsigned int);
+    float (__cdecl* pre_process_get_delay)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE, unsigned int);
+    unsigned int (__cdecl* pre_process)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE, unsigned int, float**, float**, unsigned int);
+
+    AE_DSP_ERROR (__cdecl* master_process_set_mode)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE, AE_DSP_STREAMTYPE, unsigned int, int);
+    unsigned int (__cdecl* master_process_needed_samplesize)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE);
+    float (__cdecl* master_process_get_delay)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE);
+    int (__cdecl* master_process_get_out_channels)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE, unsigned long&);
+    unsigned int (__cdecl* master_process)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE, float**, float**, unsigned int);
+    const char* (__cdecl* master_process_get_stream_info_string)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE);
+
+    unsigned int (__cdecl* post_process_needed_samplesize)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE, unsigned int);
+    float (__cdecl* post_process_get_delay)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE, unsigned int);
+    unsigned int (__cdecl* post_process)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE, unsigned int, float**, float**, unsigned int);
+
+    unsigned int (__cdecl* output_resample_process_needed_samplesize)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE);
+    unsigned int (__cdecl* output_resample_process)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE, float**, float**, unsigned int);
+    float (__cdecl* output_resample_get_delay)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE);
+    int (__cdecl* output_resample_samplerate)(AddonInstance_AudioDSP* addonInstance, const ADDON_HANDLE);
+  } KodiToAddonFuncTable_AudioDSP;
+
+  typedef struct AddonInstance_AudioDSP
+  {
+    AddonProps_AudioDSP props;
+    AddonToKodiFuncTable_AudioDSP toKodi;
+    KodiToAddonFuncTable_AudioDSP toAddon;
+  } AddonInstance_AudioDSP;
+
+} /* extern "C" */
+
+namespace kodi {
+namespace addon {
+
+  class CInstanceAudioDSP : public IAddonInstance
+  {
+  public:
+    //==========================================================================
+    /// @brief Class constructor
+    ///
+    CInstanceAudioDSP()
+      : IAddonInstance(ADDON_INSTANCE_ADSP)
+    {
+      if (CAddonBase::m_interface->globalSingleInstance != nullptr)
+        throw std::logic_error("kodi::addon::CInstanceAudioDSP: Creation of more as one in single instance way is not allowed!");
```

Comment:
With `throw` is it not direct possible. Only can it then done with a:
```cpp
std::logic_error(StringUtils::Format("kodi::addon::%s: Creation of more as one in single instance way is not allowed!", __FUNCTION__).c_str())
```
But this is for a part who normally never come a bit much :)